### PR TITLE
add .gitignore

### PR DESCRIPTION
**Pull request by afinch7**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+target
```
Comment (kubkon) [inline .gitignore]: If you could change that to `/target` to be compatible with Cargo defaults, that'd be great.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+Cargo.lock
+.*.rustfmt
```
Comment (kubkon) [inline .gitignore]: I'm not sure that's actually needed. However, instead, if you could add `**/*.rs.bk` to be compatible with Cargo defaults, that'd be awesome!

**Comment by kubkon** [comment]
Hey! Thanks for this! It looks good, just a couple of minor nitpicking.

**Comment by kubkon** [comment]
Looks good, thanks! Just one more thing, could you explain why you've added `.*.rustfmt`? I can't seem to be able to find a use case for it with my setup.

**Comment by afinch7** [comment]
I was looking at the .gitignore in wasmtime for reference. I wasn't sure why it was there either, but it looks like it was added a long time ago in wasmtime.

**Comment by kubkon** [comment]
Ah, ok, good to know! I'm not sure why it's there either. Perhaps @sunfishcode could shed some light on this?

**Comment by sunfishcode** [comment]
I'm not sure why it's there either, offhand. Removing it is fine with me!

**Comment by kubkon** [comment]
LGTM, thanks!

**Pull request merged**